org.aposin.licensescout.core/src/test/java/org/aposin/licensescout/finder/JavaJarFinderTest.java

```diff
@@ -43,6 +62,355 @@ public void testPomResolutionUsed() {
+        final FinderResult finderResult = doScan(scanDirectoryPacked);
+        assertSingleArchive(finderResult, scanDirectoryPacked, true, true, 1);
+    }
+
+    /**
+     * Test method for {@link JavaJarFinder#findLicenses())}.
+     * 
+     * <p>Tests with a packed JAR and license information in the POM.</p>
+     * <p>Uses scan directory 'src/test/resources/scans/java-packed-license-included-pom'.</p>
+     * 
+     * @throws Exception 
+     */
+    @Test
+    public void testScanJavaPackedLicenseIncludedPom() throws Exception {
+        final File scanDirectoryUnpacked = getFile("java-unpacked-license-included-pom/lib1");
+        final File scanDirectoryPacked = getFile("java-packed-license-included-pom");
+        createJarLib1(scanDirectoryUnpacked, scanDirectoryPacked);
+        final FinderResult finderResult = doScan(scanDirectoryPacked);
+        assertSingleArchive(finderResult, scanDirectoryPacked, true, true, 0);
+    }
+
+    /**
+     * Test method for {@link JavaJarFinder#findLicenses())}.
+     * 
+     * <p>Tests with a packed JAR and license information in the MANIFEST.MF file.</p>
+     * <p>Uses scan directory 'src/test/resources/scans/java-packed-license-manifest'.</p>
+     * 
+     * @throws Exception 
+     */
+    @Test
+    public void testScanJavaPackedLicenseManifest() throws Exception {
+        final File scanDirectoryUnpacked = getFile("java-unpacked-license-manifest/lib1");
+        final File scanDirectoryPacked = getFile("java-packed-license-manifest");
+        createJarLib1(scanDirectoryUnpacked, scanDirectoryPacked);
+        final FinderResult finderResult = doScan(scanDirectoryPacked);
+        assertSingleArchive(finderResult, scanDirectoryPacked, true, true, 0);
+    }
+
+    /**
+     * Test method for {@link JavaJarFinder#findLicenses())}.
+     * 
+     * <p>Tests with a packed JAR and no license information.</p>
+     * <p>Uses scan directory 'src/test/resources/scans/java-packed-license-none'.</p>
+     * 
+     * @throws Exception 
+     */
+    @Test
+    public void testScanJavaPackedLicenseNone() throws Exception {
+        final File scanDirectoryUnpacked = getFile("java-unpacked-license-none/lib1");
+        final File scanDirectoryPacked = getFile("java-packed-license-none");
+        createJarLib1(scanDirectoryUnpacked, scanDirectoryPacked);
+        final FinderResult finderResult = doScan(scanDirectoryPacked);
+        assertSingleArchive(finderResult, scanDirectoryPacked, true, false, 0);
+    }
+
+    /**
+     * Test method for {@link JavaJarFinder#findLicenses())}.
+     * 
+     * <p>Tests with a packed JAR without license information and
+     * an inner packed JAR with a license file.</p>
+     * <p>Uses scan directory 'src/test/resources/scans/java-packed-with-inner-license-file'.</p>
+     * 
+     * @throws Exception 
+     */
+    @Test
+    public void testScanJavaPackedWithInnerLicenseFile() throws Exception {
+        {
```

Comment:
[Optional] can we get rid of those brackets (here and in the following tests as well)?

Reply:
Changed.